src/scenes/Modals/ConfirmLeaveModal.tsx

```diff
@@ -9,8 +9,35 @@ import useBackButton from 'hooks/useBackButton';
 import { useFragment } from 'react-relay';
 import { graphql } from 'relay-runtime';
 import { ConfirmLeaveModalFragment$key } from '__generated__/ConfirmLeaveModalFragment.graphql';
+import { useRouter } from 'next/router';
 
-function ConfirmLeaveModal({ userRef }: { userRef: ConfirmLeaveModalFragment$key }) {
+function Modal({ onClick }: { onClick: () => void }) {
+  return (
+    <StyledConfirmLeaveModal>
+      <LeaveWrapper>
+        <BaseM>Would you like to stop editing?</BaseM>
+        <Spacer height={28}></Spacer>
+        <StyledButton onClick={onClick} text={'Leave'} />
+      </LeaveWrapper>
+    </StyledConfirmLeaveModal>
+  );
+}
+
+// A generic modal that simply "goes back" in history after user confirms
+export function ConfirmBackModal() {
+  const { hideModal } = useModalActions();
+  const { back } = useRouter();
+
+  const goBack = useCallback(() => {
+    back();
+    hideModal();
+  }, [back, hideModal]);
+
+  return <Modal onClick={goBack} />;
+}
```

Comment:
I've split up `ConfirmLeaveModal.tsx` to include two components:
* One `ConfirmBackModal` which just runs the generic `back()` from `useRouter` when the user confirms (this mimics the behavior of the back button in the collection editor currently)
* `ConfirmLeaveModal` which contains the back functionality used elsewhere.

`ConfirmBackModal` does not require a username. It simply runs `back()`, which is exactly what clicking the back button did in the old configuration ([see here](https://github.com/gallery-so/gallery/pull/808/files#diff-993ecc44e51efddeff84df7b02d4730e1db35f41bf45645479b0273536572d28L70-L72)).

Tl;dr, this just wraps the `back()` functionality in a modal